lib/wasix/src/bin_factory/exec.rs

```diff
@@ -54,25 +53,36 @@ pub async fn spawn_exec_wasm(
     spawn_exec_module(module, env, runtime)
 }
 
-pub async fn spawn_load_wasm<'a>(
-    binary: &'a BinaryPackage,
+pub fn package_command_by_name<'a>(
+    pkg: &'a BinaryPackage,
     name: &str,
-) -> Result<&'a [u8], SpawnError> {
-    let wasm = if let Some(cmd) = binary.get_command(name) {
-        cmd.atom.as_ref()
-    } else if let Some(cmd) = binary.get_entrypoint_command() {
-        &cmd.atom
+) -> Result<&'a BinaryPackageCommand, SpawnError> {
+    // If an explicit command is provided, use it.
+    // Otherwise, use the entrypoint.
+    // If no entrypoint exists, and the package has a single
+    // command, then use it. This is done for backwards
+    // compatibility.
+    let cmd = if let Some(cmd) = pkg.get_command(&name) {
+        cmd
+    } else if let Some(cmd) = pkg.get_entrypoint_command() {
+        cmd
     } else {
-        tracing::error!(
-          command=name,
-          pkg=%binary.id,
-          "Unable to spawn a command because its package has no entrypoint",
-        );
-        return Err(SpawnError::MissingEntrypoint {
-            package_id: binary.id.clone(),
-        });
+        match pkg.commands.as_slice() {
+            [] => {
+                return Err(SpawnError::MissingEntrypoint {
+                    package_id: pkg.id.clone(),
+                });
+            }
+            [first] => first,
+            [_first, ..] => {
```

Comment:
This time around, I'm not approving this 😄 this case can be merged with the empty slice case.

Reply:
Buuut it's supposed to signal a difference between "no commands" and "multiple commands, don't know which to pick".

We can extend the error with a message to reflect this (that also mentions the command names).